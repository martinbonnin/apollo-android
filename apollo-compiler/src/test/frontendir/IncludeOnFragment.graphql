query GetAnimal($condition: Boolean!) {
  randomAnimal {
    species
    ...catFragment @include(if: $condition)
  }
}

fragment catFragment on Cat {
  name
  habitat {
    temperature
  }
}
