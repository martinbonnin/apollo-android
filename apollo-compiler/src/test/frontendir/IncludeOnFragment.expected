package com.example

import kotlin.Float
import kotlin.String

interface GetAnimal {
  interface Query {
    val randomAnimal: Animal?

    interface AnimalCatCondition {
      val species: String

      val name: String

      val habitat: com.example.Habitat

      interface Habitat {
        val temperature: Float?
      }
    }

    interface AnimalCatCondition {
      val species: String
    }
  }
}

interface CatFragment {
  interface Cat {
    val name: String

    val habitat: com.example.Habitat

    interface Habitat {
      val temperature: Float?
    }
  }
}
